youths/schema.py

```diff
@@ -0,0 +1,194 @@
+from datetime import datetime
+
+import graphene
+from django.core import signing
+from django.core.exceptions import ObjectDoesNotExist
+from django_ilmoitin.utils import send_notification
+from graphene_django.types import DjangoObjectType
+from graphql_jwt.decorators import login_required
+
+from profiles.models import Profile
+
+from .consts import GENDERS, LANGUAGES
+from .models import YouthProfile
+from .utils import create_approval_token, mail_encrypted_approval_link
+
+
+class YouthProfileType(DjangoObjectType):
+    class Meta:
+        model = YouthProfile
+        exclude = ["id", "approval_token"]
+
+
+PreferredLanguage = graphene.Enum(
+    "PreferredLanguage", [(l[0].upper(), l[0]) for l in LANGUAGES]
+)
+Gender = graphene.Enum("Gender", [(g[0].upper(), g[0]) for g in GENDERS])
+
+
+# Abstract base fields
+class YouthProfileFields(object):
+    school_name = graphene.String()
+    school_class = graphene.String()
+    preferred_language = PreferredLanguage()
+    volunteer_info = graphene.String()
+    gender = Gender()
+    diabetes = graphene.Boolean()
+    epilepsy = graphene.Boolean()
+    heart_disease = graphene.Boolean()
+    extra_illnesses_info = graphene.String()
+    serious_allergies = graphene.Boolean()
+    allergies = graphene.String()
+    notes = graphene.String()
+    approver_email = graphene.String()
+
+
+# Subset of abstract fields are required for creation
+class CreateYouthProfileInput(YouthProfileFields, graphene.InputObjectType):
+    ssn = graphene.String(required=True)
+    school_name = graphene.String(required=True)
+    school_class = graphene.String(required=True)
+    approver_email = graphene.String(required=True)
+
+
+class CreateYouthProfile(graphene.Mutation):
+    class Arguments:
+        youth_profile_data = CreateYouthProfileInput(required=True)
+        profile_id = graphene.Int()
+
+    youth_profile = graphene.Field(YouthProfileType)
+
+    @login_required
+    def mutate(self, info, **kwargs):
+        input_data = kwargs.pop("youth_profile_data")
+        if (
+            info.context.user.is_superuser
+        ):  # FIXME should be something like user.is_staff_or_more?
+            profile = Profile.objects.get(pk=kwargs.pop("profile_id"))
```

Comment:
`kwargs.pop("profile_id")` might resolve to `None` and the `profile` will be `None` too. later on fetching `YouthProfile` will fail, if no profile passed, I suppose. maybe raise an exception, if admin is trying to update someone's profile without passing its `id`?

Reply:
`from graphql import GraphQLError`